rest/changes_api.go

```diff
@@ -354,6 +357,25 @@ func (h *handler) sendChangesForDocIds(userChannels base.Set, explicitDocIds []s
 					}
 				}
 			}
+		} else if h.user.Channels().Contains(ch.UserStarChannel) {
+			userCanSeeDocChannel = true
+		}
```

Comment:
If the user has the star channel, we can skip the iteration over the doc's channels.  So maybe flip this if statement to 'if has star channel, else doc.channels > 0?